lib/android/src/main/java/com/wix/reactnativekeyboardinput/CustomKeyboardLayout.java

```diff
@@ -95,33 +94,45 @@ public void run() {
     }
 
     private void showCustomKeyboardContent() {
-        dispatchUIUpdates(new Runnable() {
+        final int expendedHeight = getHeightForCustomContent();
+        if (!setCustomKeyboardHeight(expendedHeight)) {
+            syncHeightWithShadowNode(expendedHeight);
+        }
+    }
+
+    private void hideCustomKeyboardContent() {
+        if (!setCustomKeyboardHeight(0)) {
+            syncHeightWithShadowNode(0);
+        }
+        runOnUIThread(new Runnable() {
             @Override
             public void run() {
-                final CustomKeyboardRootViewShadow shadowNode = mShadowNode.get();
-                if (shadowNode != null) {
-                    shadowNode.setHeight(getHeightForCustomContent());
-                }
+                sendCustomKeyboardResignedEvent();
             }
         });
     }
 
-    private void hideCustomKeyboardContent() {
+    private void syncHeightWithShadowNode(final int height) {
         dispatchUIUpdates(new Runnable() {
             @Override
             public void run() {
-                final CustomKeyboardRootViewShadow shadowNode = mShadowNode.get();
-                if (shadowNode != null) {
-                    shadowNode.setHeight(0);
-                }
+                setCustomKeyboardHeight(height);
             }
         });
-        runOnUIThread(new Runnable() {
-            @Override
-            public void run() {
-                sendCustomKeyboardResignedEvent();
+    }
+
+    private boolean setCustomKeyboardHeight(int height) {
```

Comment:
How about `setCustomKeyboardHeightImmediate` or smt?

Reply:
And then also `syncHeightWithShadowNode()` => `syncCustomKeyboardHeight()` so it would be clear they're similar